File: friend-integration/poster_link.py
"""Server-side helper for a Python scheduling website."""
import json
import os
import urllib.request


def create_poster_link(schedule):
    request = urllib.request.Request(
        os.environ["POSTER_SERVICE_URL"].rstrip("/") + "/api/create-poster-link",
        data=json.dumps({"schedule": schedule, "auto_generate": True}).encode("utf-8"),
        method="POST",
        headers={
            "Content-Type": "application/json",
            "Authorization": "Bearer " + os.environ["POSTER_INTEGRATION_SECRET"],
        },
    )
    with urllib.request.urlopen(request, timeout=30) as response:
        return json.loads(response.read().decode("utf-8"))["url"]


# Flask example:
# return redirect(create_poster_link(saved_schedule))

File: friend-integration/poster-link.js
/**
 * SERVER-SIDE example for your friend's scheduling website.
 * Never run this code in browser JavaScript because it uses the integration secret.
 */
export async function createPosterLink(schedule) {
  const response = await fetch(`${process.env.POSTER_SERVICE_URL}/api/create-poster-link`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      Authorization: `Bearer ${process.env.POSTER_INTEGRATION_SECRET}`
    },
    body: JSON.stringify({ schedule, auto_generate: true })
  });
  const data = await response.json();
  if (!response.ok) throw new Error(data.error || 'Could not create poster link');
  return data.url;
}

// In the schedule website's authenticated button handler:
// const posterUrl = await createPosterLink(savedSchedule);
// response.redirect(posterUrl);
